doc/api.rst

```diff
@@ -100,8 +100,20 @@ Parallel backends (:py:mod:`hnn_core.parallel_backends`):
    MPIBackend
    JoblibBackend
 
-Input and Output:
------------------
+Input and Output (hierarchical JSON) (:py:mod:`hnn_core.hnn_io`):
+-----------------------------------------------------------------
+.. currentmodule:: hnn_core.hnn_io
+
+.. autosummary::
+   :toctree: generated/
+
+   dict_to_network
+   network_to_dict
+   read_network_configuration
+   write_network_configuration
```

Comment:
I haven't closely followed the developments but out of curiosity, what is the difference between `network_to_dict` and `write_network_configuration`? Could the user just write a json from the dictionary? You potentially want to avoid too many ways to do the same thing. Zen of Python:

> There should be one-- and preferably only one --obvious way to do it.